rest-service/manager_rest/resource_manager.py

```diff
@@ -107,6 +108,73 @@ def list_executions(self, include=None, is_include_system_workflows=False,
             get_all_results=get_all_results
         )
 
+    @staticmethod
+    def evaluate_summary_deployment_status(latest_execution_status,
+                                           installation_status):
+        """
+        Evaluate the overall deployment status based on installation status
+        and latest execution object
+        :param latest_execution_status: The latest execution status for
+        deployment
+        :param installation_status: The installation status for deployment
+        :return: deployment_status: Overall deployment status
+        """
+        # TODO we need to cover also aggregated statuses for deployment
+        #  children later on
+        if latest_execution_status == DeploymentState.CANCELLED:
+            if installation_status == DeploymentState.ACTIVE:
+                return DeploymentState.GOOD
+            return DeploymentState.REQUIRE_ATTENTION
+        elif latest_execution_status == DeploymentState.IN_PROGRESS and \
+                installation_status == DeploymentState.INACTIVE:
+            return DeploymentState.IN_PROGRESS
+        elif latest_execution_status == DeploymentState.FAILED or \
+                installation_status == DeploymentState.INACTIVE:
+            return DeploymentState.REQUIRE_ATTENTION
+        elif latest_execution_status == DeploymentState.COMPLETED and \
+                installation_status == DeploymentState.ACTIVE:
+            return DeploymentState.GOOD
+        else:
+            return DeploymentState.IN_PROGRESS
+
+    def update_deployment_statuses(self, latest_execution):
+        """
+        Update deployment statuses based on latest execution
+        :param latest_execution: Latest execution object
+        """
+        installation_status = DeploymentState.INACTIVE
+        if not (latest_execution and latest_execution.deployment_id):
+            return
+
+        node_instances = self.sm.list(models.NodeInstance, filters={
+            'deployment_id': latest_execution.deployment_id
+        })
+        if node_instances:
+            total_node_instances = len(node_instances)
+            started_node_instances = 0
+            for node_instance in node_instances:
+                if node_instance.state == ExecutionState.STARTED:
+                    started_node_instances += 1
+
+            if total_node_instances == started_node_instances:
+                installation_status = DeploymentState.ACTIVE
```

Comment:
```suggestion
installation_status = DeploymentState.ACTIVE
for node_instance in node_instances:
    if node_instance.state != ExecutionState.STARTED:
        installation_status = DeploymentState.INACTIVE
        break       
```

Reply:
the `if` statement can be deleted sure will update that